ami/ml/schemas.py

```diff
@@ -62,6 +62,65 @@ class PipelineRequest(pydantic.BaseModel):
 class PipelineResponse(pydantic.BaseModel):
     # pipeline: PipelineChoice
     pipeline: str
-    total_time: float
+    total_time: float | None
     source_images: list[SourceImageResponse]
     detections: list[DetectionResponse]
+    errors: list | str | None = None
+
+
+class PipelineStageParam(pydantic.BaseModel):
+    """A configurable parameter of a stage of a pipeline."""
+
+    name: str
+    key: str
+    category: str = "default"
+
+
+class PipelineStage(pydantic.BaseModel):
+    """A configurable stage of a pipeline."""
+
+    key: str
+    name: str
+    params: list[PipelineStageParam] = []
+    description: str | None = None
+
+
+class ProjectConfig(pydantic.BaseModel):
+    name: str
+
+
+class AlgorithmConfig(pydantic.BaseModel):
+    name: str
+    key: str
+
+
+class PipelineConfig(pydantic.BaseModel):
+    """A configurable pipeline."""
+
+    name: str
+    slug: str
+    version: int
+    description: str | None = None
+    algorithms: list[AlgorithmConfig] = []
+    stages: list[PipelineStage] = []
+    projects: list[ProjectConfig] = []
```

Comment:
The ML backend will never know about the projects configured in Antenna, so you can remove projects here. These schemas are only for the ML backend requests & responses. The serializers are for the Antenna project & models. Sorry its confusing!